pkg/error/error_test.go

```diff
@@ -112,6 +113,69 @@ func TestGetErrorCount(t *testing.T) {
 	}
 }
 
+func TestValidationErrorEqual(t *testing.T) {
+	baseError := ValidationError{
+		LineNumber: -1,
+		Message:    errors.New("a message"),
+	}
+	wrongLineNumberError := ValidationError{
+		LineNumber: 2,
+		Message:    errors.New("a message"),
+	}
+	differentMessageError := ValidationError{
+		LineNumber: -1,
+		Message:    errors.New("different message"),
+	}
+	differentCountError := ValidationError{
+		LineNumber:       -1,
+		Message:          errors.New("a message"),
+		ConsecutiveCount: 1,
+	}
+	if !baseError.Equal(baseError) {
+		t.Error("failed to detect an error being equal to itself")
+	}
+	if baseError.Equal(wrongLineNumberError) {
+		t.Error("failed to detect a difference in the LineNumber")
+	}
+	if baseError.Equal(differentMessageError) {
+		t.Error("failed to detect a difference in the Message")
+	}
+	if baseError.Equal(differentCountError) {
+		t.Error("failed to detect a difference in the ConsequtiveCount")
+	}
+}
+
+func TestConsolidateErrors(t *testing.T) {
+	input := []ValidationError{
+		// two messages that become one block
+		{LineNumber: 1, Message: errors.New("message kind one")},
+		{LineNumber: 2, Message: errors.New("message kind one")},
+		// one message with a good line between it and the last bad line, but repeating the message
+		{LineNumber: 4, Message: errors.New("message kind one")},
+		// one message that breaks the continousnes of line 4 to line 6
+		{LineNumber: 5, Message: errors.New("message kind two")},
+		{LineNumber: 6, Message: errors.New("message kind one")},
+		// one message without a line number, that will become sorted to the top
+		{LineNumber: -1, Message: errors.New("file-level error")},
+	}
+
+	expected := []ValidationError{
+		{LineNumber: -1, Message: errors.New("file-level error")},
+		{LineNumber: 1, ConsecutiveCount: 1, Message: errors.New("message kind one")},
```

Comment:
I know lists start at 0

But I would have expected to find a Consecutive count of 2.

With the current example and code, line 1 will report a consecutive count of 1.

A human would expect to know that the error reported line 1 was reported twice, so 2

Reply:
hm.. ConsecutiveCount is the count of additional error messages that followed the first one - so it excludes the first one (meaning `ConsecutiveCount == 0` is "one singular message").
But switching to storing according to your model would mean the last-line-of-block calculation becomes a tad simpler. I'll think about that one - maybe renaming the property to a better name is  better to make it clearer (I'm thinking about `AdditionalIdenticalErrorCount` for example)

Reply:
It would make sense.

I'm not good with wording, but yes I think it would make it easier to understand.

Reply:
I renamed it, sadly I had no better idea than `AdditionalIdenticalErrorCount`, so that one it is for  now ;)

Reply:
Someone may come and suggest a new name

Reply:
I'm open for suggestions - a rename is cheap!